src/TreeHouse/QueueBundle/DependencyInjection/TreeHouseQueueExtension.php

```diff
@@ -31,6 +34,7 @@ public function load(array $configs, ContainerBuilder $container)
         $this->loadConnections($config, $container);
```

Comment:
Maybe an option to introduce a simple factory pattern to abstract all the 'loading'.

In this case load could consist of several objects:
- Connections
- Publishers
- Exchanges
- Queues

https://github.com/domnikl/DesignPatternsPHP/tree/master/Creational/SimpleFactory